### ISSUE TITLE
Make Docker build O/S dependencies

### ISSUE THREAD

**Issue by karianna** [AdoptOpenJDK/openjdk-build#1271]
@Willsparker for MAINTAINER - https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

The other issue.... hmm it's a good point.  Ideally, we should be pinning all of those versions to get repeatable builds.  Tell you what.  I'll open up a separate issue called "Make Docker build O/S dependencies" so we can fix that all later.

So just the MAINTAINER fix and we can merge this. :-)

_Originally posted by @karianna in https://github.com/AdoptOpenJDK/openjdk-build/pull/1256#issuecomment-529904840_

**Comment by sxa555** [comment]
ping @Willsparker - anything outstanding on this?

**Comment by Willsparker** [comment]
I don't believe anything has been done about this. The only thing that has pinned versions is `GCC/G++ 7` in the openj9 dockerfiles. I can pin them to be the versions they are currently, as the DockerfileCheck Jenkins Job is looking good right now: https://ci.adoptopenjdk.net/job/DockerfileCheck/85/

**Comment by sxa555** [comment]
@karianna What's your view on this? In general we build with whatever the versions on any given distribution have as latest - do you believe the need to be locked down further than that (thus potentially using insecure versions) - I'm not sure why we explicitly want that enforcement in the dockerfiles though

**Comment by karianna** [comment]
My view is that we should be building with stable versions of libs (repeatable builds) but updating aggressively in the case of CVEs.  That is I'd expect we should be able to roll out docker images more often than the quarterly Java CPU should we find an underlying base CVE.

**Comment by sxa555** [comment]
That would like require us to keep quite a close eye on which packages have vulnerabilities and manually update each of those as and when required. I'm not sure that's something we realistically have the resources to take on for all the distributions we support ... On that basis I'd find it hard to recommend I would personally be more willing to "trust" the distribution's update policy in that respect unless we have a volunteer to keep this stuff updated - I'd personally rather "over-update" then "under-update" by locking things down and not having them updated for CVEs
